<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SPMController</class>
 <widget class="QWidget" name="SPMController">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1120</width>
    <height>900</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>SPMController</string>
  </property>
  <widget class="QLabel" name="label_all_range">
   <property name="geometry">
    <rect>
     <x>120</x>
     <y>30</y>
     <width>321</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:18pt;&quot;&gt;Select Scan Range Within Peizo Limits&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_scan_window_ch1">
   <property name="geometry">
    <rect>
     <x>130</x>
     <y>510</y>
     <width>301</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:18pt;&quot;&gt;Image within scan window Channel 1&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="doubleSpinBox_piezo_limit_x">
   <property name="geometry">
    <rect>
     <x>120</x>
     <y>100</y>
     <width>62</width>
     <height>22</height>
    </rect>
   </property>
   <property name="keyboardTracking">
    <bool>false</bool>
   </property>
   <property name="value">
    <double>50.000000000000000</double>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="doubleSpinBox_piezo_limit_y">
   <property name="geometry">
    <rect>
     <x>360</x>
     <y>440</y>
     <width>62</width>
     <height>22</height>
    </rect>
   </property>
   <property name="keyboardTracking">
    <bool>false</bool>
   </property>
   <property name="value">
    <double>50.000000000000000</double>
   </property>
  </widget>
  <widget class="QLabel" name="label_piezo_limit_x">
   <property name="geometry">
    <rect>
     <x>120</x>
     <y>60</y>
     <width>121</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>Piezo Limit x (V)
Recommended: 50</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_piezo_limit_y">
   <property name="geometry">
    <rect>
     <x>360</x>
     <y>470</y>
     <width>121</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>Piezo Limit y (V)
Recommended: 50</string>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="doubleSpinBox_x_min">
   <property name="geometry">
    <rect>
     <x>700</x>
     <y>70</y>
     <width>81</width>
     <height>22</height>
    </rect>
   </property>
   <property name="keyboardTracking">
    <bool>false</bool>
   </property>
   <property name="maximum">
    <double>99.989999999999995</double>
   </property>
  </widget>
  <widget class="QLabel" name="label_x_scan_direction">
   <property name="geometry">
    <rect>
     <x>270</x>
     <y>110</y>
     <width>121</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; color:#ff9300;&quot;&gt;&amp;lt;------- x -------&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </widget>
  <widget class="QLabel" name="labelx_y_scan_direction">
   <property name="geometry">
    <rect>
     <x>430</x>
     <y>120</y>
     <width>20</width>
     <height>131</height>
    </rect>
   </property>
   <property name="text">
    <string> |
 |
 |
 y
 |
 |
↓</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_position_0_0">
   <property name="geometry">
    <rect>
     <x>380</x>
     <y>110</y>
     <width>91</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; color:#ff2600;&quot;&gt;Position (0, 0)&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_note">
   <property name="geometry">
    <rect>
     <x>370</x>
     <y>70</y>
     <width>161</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;Note: Scan window near (0, 0) &lt;br/&gt;has smallest piezo distortion&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="doubleSpinBox_x_max">
   <property name="geometry">
    <rect>
     <x>700</x>
     <y>110</y>
     <width>81</width>
     <height>21</height>
    </rect>
   </property>
   <property name="keyboardTracking">
    <bool>false</bool>
   </property>
   <property name="maximum">
    <double>99.989999999999995</double>
   </property>
   <property name="value">
    <double>20.000000000000000</double>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="doubleSpinBox_y_max">
   <property name="geometry">
    <rect>
     <x>900</x>
     <y>110</y>
     <width>81</width>
     <height>22</height>
    </rect>
   </property>
   <property name="keyboardTracking">
    <bool>false</bool>
   </property>
   <property name="value">
    <double>10.000000000000000</double>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="doubleSpinBox_y_min">
   <property name="geometry">
    <rect>
     <x>900</x>
     <y>70</y>
     <width>81</width>
     <height>22</height>
    </rect>
   </property>
   <property name="keyboardTracking">
    <bool>false</bool>
   </property>
  </widget>
  <widget class="QLabel" name="label_x_min">
   <property name="geometry">
    <rect>
     <x>630</x>
     <y>70</y>
     <width>60</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>x min (V)</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_x_min_2">
   <property name="geometry">
    <rect>
     <x>630</x>
     <y>110</y>
     <width>60</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>x max (V)</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_y_min">
   <property name="geometry">
    <rect>
     <x>830</x>
     <y>70</y>
     <width>60</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>y min (V)</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_y_max">
   <property name="geometry">
    <rect>
     <x>830</x>
     <y>110</y>
     <width>60</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>y max (V)</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_note2">
   <property name="geometry">
    <rect>
     <x>610</x>
     <y>80</y>
     <width>91</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;Recommended: 0&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_note3">
   <property name="geometry">
    <rect>
     <x>610</x>
     <y>120</y>
     <width>91</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;Recommended: 0&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </widget>
  <widget class="QSpinBox" name="spinBox_x_pixels">
   <property name="geometry">
    <rect>
     <x>700</x>
     <y>150</y>
     <width>81</width>
     <height>21</height>
    </rect>
   </property>
   <property name="keyboardTracking">
    <bool>false</bool>
   </property>
   <property name="maximum">
    <number>1000000000</number>
   </property>
   <property name="value">
    <number>256</number>
   </property>
  </widget>
  <widget class="QSpinBox" name="spinBox_y_pixels">
   <property name="geometry">
    <rect>
     <x>900</x>
     <y>150</y>
     <width>81</width>
     <height>22</height>
    </rect>
   </property>
   <property name="keyboardTracking">
    <bool>false</bool>
   </property>
   <property name="maximum">
    <number>1000000000</number>
   </property>
   <property name="value">
    <number>256</number>
   </property>
  </widget>
  <widget class="QLabel" name="label_x_pixels">
   <property name="geometry">
    <rect>
     <x>630</x>
     <y>150</y>
     <width>51</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>x pixels</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_y_pixels">
   <property name="geometry">
    <rect>
     <x>830</x>
     <y>150</y>
     <width>51</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>y pixels</string>
   </property>
  </widget>
  <widget class="PlotWidget" name="widget_linescan_ch1" native="true">
   <property name="geometry">
    <rect>
     <x>580</x>
     <y>260</y>
     <width>441</width>
     <height>101</height>
    </rect>
   </property>
   <property name="autoFillBackground">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="PlotWidget" name="widget_linescan_ch2" native="true">
   <property name="geometry">
    <rect>
     <x>580</x>
     <y>390</y>
     <width>441</width>
     <height>101</height>
    </rect>
   </property>
   <property name="autoFillBackground">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QRadioButton" name="radioButton_ch1">
   <property name="geometry">
    <rect>
     <x>120</x>
     <y>430</y>
     <width>161</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>Show Channel 1</string>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QRadioButton" name="radioButton_ch2">
   <property name="geometry">
    <rect>
     <x>120</x>
     <y>450</y>
     <width>131</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>Show Channel 2</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_scan_window_ch2">
   <property name="geometry">
    <rect>
     <x>580</x>
     <y>510</y>
     <width>301</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:18pt;&quot;&gt;Image within scan window Channel 2&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_linescan_ch1">
   <property name="geometry">
    <rect>
     <x>580</x>
     <y>230</y>
     <width>131</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Line scan Channel 1</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_linescan_ch2">
   <property name="geometry">
    <rect>
     <x>580</x>
     <y>370</y>
     <width>131</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Line scan Channel 2</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_frequency">
   <property name="geometry">
    <rect>
     <x>589</x>
     <y>190</y>
     <width>101</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>Frequency (Hz)</string>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="doubleSpinBox_rotation">
   <property name="geometry">
    <rect>
     <x>900</x>
     <y>190</y>
     <width>81</width>
     <height>21</height>
    </rect>
   </property>
   <property name="keyboardTracking">
    <bool>false</bool>
   </property>
   <property name="minimum">
    <double>-180.000000000000000</double>
   </property>
   <property name="maximum">
    <double>180.000000000000000</double>
   </property>
   <property name="value">
    <double>0.000000000000000</double>
   </property>
  </widget>
  <widget class="QLabel" name="label_rotation">
   <property name="geometry">
    <rect>
     <x>810</x>
     <y>190</y>
     <width>81</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>Rotation (°)</string>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="doubleSpinBox_frequency">
   <property name="geometry">
    <rect>
     <x>700</x>
     <y>190</y>
     <width>81</width>
     <height>22</height>
    </rect>
   </property>
   <property name="keyboardTracking">
    <bool>false</bool>
   </property>
   <property name="maximum">
    <double>1.000000000000000</double>
   </property>
   <property name="value">
    <double>0.020000000000000</double>
   </property>
  </widget>
  <widget class="QPushButton" name="pushButton_scan">
   <property name="geometry">
    <rect>
     <x>720</x>
     <y>30</y>
     <width>71</width>
     <height>32</height>
    </rect>
   </property>
   <property name="text">
    <string>Scan</string>
   </property>
  </widget>
  <widget class="QPushButton" name="pushButton_image">
   <property name="geometry">
    <rect>
     <x>810</x>
     <y>30</y>
     <width>71</width>
     <height>32</height>
    </rect>
   </property>
   <property name="text">
    <string>Image</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_filename_ch1">
   <property name="geometry">
    <rect>
     <x>430</x>
     <y>690</y>
     <width>111</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string>Channel 1
Filenames (txt, tiff)
trace and retrace</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_filename_ch2">
   <property name="geometry">
    <rect>
     <x>880</x>
     <y>680</y>
     <width>111</width>
     <height>61</height>
    </rect>
   </property>
   <property name="text">
    <string>Channel 2
Filenames (txt, tiff)
trace and retrace</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_directory">
   <property name="geometry">
    <rect>
     <x>120</x>
     <y>860</y>
     <width>101</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Save Directory:</string>
   </property>
  </widget>
  <widget class="QPushButton" name="pushButton_directory">
   <property name="geometry">
    <rect>
     <x>860</x>
     <y>850</y>
     <width>121</width>
     <height>32</height>
    </rect>
   </property>
   <property name="text">
    <string>Select Directory</string>
   </property>
  </widget>
  <widget class="QWidget" name="verticalLayoutWidget">
   <property name="geometry">
    <rect>
     <x>120</x>
     <y>130</y>
     <width>301</width>
     <height>301</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_display_piezo_limit"/>
  </widget>
  <widget class="QWidget" name="verticalLayoutWidget_2">
   <property name="geometry">
    <rect>
     <x>120</x>
     <y>540</y>
     <width>301</width>
     <height>301</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_display_scan_window_ch1"/>
  </widget>
  <widget class="QWidget" name="verticalLayoutWidget_3">
   <property name="geometry">
    <rect>
     <x>570</x>
     <y>540</y>
     <width>301</width>
     <height>301</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_display_scan_window_ch2"/>
  </widget>
  <widget class="QLabel" name="label_rotation_2">
   <property name="geometry">
    <rect>
     <x>810</x>
     <y>210</y>
     <width>161</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;(relative to scan window center)&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_error">
   <property name="geometry">
    <rect>
     <x>610</x>
     <y>10</y>
     <width>481</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="Line" name="line_x_y">
   <property name="geometry">
    <rect>
     <x>790</x>
     <y>70</y>
     <width>20</width>
     <height>111</height>
    </rect>
   </property>
   <property name="orientation">
    <enum>Qt::Vertical</enum>
   </property>
  </widget>
  <widget class="QLineEdit" name="lineEdit_directory">
   <property name="geometry">
    <rect>
     <x>250</x>
     <y>860</y>
     <width>581</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QLineEdit" name="lineEdit_filename_trace_ch1">
   <property name="geometry">
    <rect>
     <x>430</x>
     <y>750</y>
     <width>131</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>ch1 trace</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="lineEdit_filename_retrace_ch1">
   <property name="geometry">
    <rect>
     <x>430</x>
     <y>780</y>
     <width>131</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>ch1 retrace</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="lineEdit_filename_trace_ch2">
   <property name="geometry">
    <rect>
     <x>880</x>
     <y>750</y>
     <width>131</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>ch2 trace</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="lineEdit_filename_retrace_ch2">
   <property name="geometry">
    <rect>
     <x>880</x>
     <y>780</y>
     <width>131</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>ch2 retrace</string>
   </property>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PlotWidget</class>
   <extends>QWidget</extends>
   <header location="global">pyqtgraph</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
